### ISSUE TITLE
[0.71.1] Android <= 10.0 Keyboard.addListener cannot receive callback

### ISSUE THREAD

**Issue by arnozhang** [facebook/react-native#36001]
### Description

Keyboard.addListener cannot receive callback on Android <= 10.0

### Version

0.71.1

### Output of `npx react-native info`

System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M1
    Memory: 118.88 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.18.1 - ~/Development/skylark/node_modules/.bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - ~/Development/skylark/node_modules/.bin/npm
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 23, 24, 26, 28, 29, 30, 31, 32, 33
      Build Tools: 26.0.2, 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0, 33.0.1
      System Images: android-19 | Google APIs ARM EABI v7a, android-22 | ARM EABI v7a, android-23 
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9477386
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 18.0.2 - /Users/arnozhang/.jenv/shims/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.1 => 0.71.1 

### Steps to reproduce

- Android system version: `<= 10.0`
- React-Native upgrade to `0.71.1`
- Activity `android:windowSoftInputMode` is `adjustResize/adjustPan`
- cannot receive Keyboard events: `keyboardDidShow/keyboardDidHide`

- **keyboardIsVisible always is true** in _ReactRootView.java_:

<img width="1142" alt="image" src="https://user-images.githubusercontent.com/2292646/215405523-49bce463-9281-42ae-ad72-a28f2fd5a9cf.png">

### Snack, code example, screenshot, or link to a repository

```java
// ReactRootView.java

private void checkForKeyboardEvents() {
    ReactRootView.this.getRootView().getWindowVisibleDisplayFrame(this.mVisibleViewArea);
    WindowInsets rootInsets = ReactRootView.this.getRootView().getRootWindowInsets();
    if (rootInsets != null) {
        WindowInsetsCompat compatRootInsets = WindowInsetsCompat.toWindowInsetsCompat(rootInsets);

        // **** BUG: variable keyboardIsVisible always true on Android <= 10.0
        boolean keyboardIsVisible = compatRootInsets.isVisible(Type.ime());

        // ...
```

**Comment by artyorsh** [comment]
I can confirm this one: after 0.71.1 upgrade, we see automation tests failing because of this. Happens on all Android 10 devices from [Browserstack list](https://www.browserstack.com/list-of-browsers-and-platforms/app_live)

**Comment by NickGerleman** [comment]
The fix for this has been cherry-picked into 0.71-stable and will be released as part of 0.71.2 later this week. https://github.com/facebook/react-native/commit/e13929eae3c9072c80afe50ea3305dc546d291df
